library/adconn.c

```diff
@@ -1094,10 +1097,21 @@ authenticate_to_directory (adcli_conn *conn)
 	                     && !adcli_conn_get_use_ldaps (conn)) {
 		mech =  "GSS-SPNEGO";
 	}
-	_adcli_info ("Using %s for SASL bind", mech);
 
-	ret = ldap_sasl_interactive_bind_s (conn->ldap, NULL, mech, NULL, NULL,
-	                                    LDAP_SASL_QUIET, sasl_interact, NULL);
+	/* Populate the userDn string based on the user and naming context */
+	strncpy(username, conn->user_name, sizeof(username) - 1);
+	username[sizeof(username) - 1] = '\0';
+	for (int i = 0; i < strlen(username); i++) {
+		if (username[i] == '@') {
+			username[i] = '\0';
+			break;
+		}
+	}
+	snprintf(userDn, sizeof(userDn), "CN=%s,CN=Users,%s", username, conn->default_naming_context);
```

Comment:
It's assured that conn->default_naming_context will never exceed 79 characters?

Reply:
correct, at most it can be us 63 characters